android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TabListReducer.kt

```diff
@@ -287,9 +287,6 @@ private fun findNewSelectedTabId(
         // We found a nearby tab, let's select it.
         nearbyTab != null -> nearbyTab.id
 
-        // If there's no private tab to select anymore then just select the last regular tab
-        isPrivate -> tabs.last().id
```

Comment:
Logically, this makes sense to me because we don't have any apps that will re-select the last regular tab when all the private tab is closed.

I was looking at the [initial PR](https://github.com/mozilla-mobile/android-components/pull/3570/files#diff-7f5282f0c4236a0a8df4b62d9154864c7b967f64a26beb132ab2314a676fa05eR135) for this change and wondering if there is still any test coverage around this. If not, perhaps it is worthwhile to add back some tests here and do another quick round of review.

Reply:
Added/updated a few tests here if you want to look again